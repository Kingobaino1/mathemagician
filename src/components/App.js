import React, { Component } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '0',
      // eslint-disable-next-line react/no-unused-state
      next: null,
      // eslint-disable-next-line react/no-unused-state
      operate: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = (btnName) => {
    console.log(btnName);
    this.setState((state) => calculate(state, btnName));
    // return this.setState((state) => { console.log(calculate(state, btnName)); });
  };

  render() {
    const { total } = this.state;
    return (
      <>
        <Display value={total} />
        <ButtonPanel handleClick={this.handleClick} />
      </>
    );
  }
}

export default App;
