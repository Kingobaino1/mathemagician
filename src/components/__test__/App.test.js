import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from '../App';

describe('Home Page', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
