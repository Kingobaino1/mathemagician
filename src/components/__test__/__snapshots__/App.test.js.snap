// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should match the snapshot 1`] = `
Array [
  <div
    className="d-flex justify-content-between"
  >
    <ul
      className="d-none d-md-block m-4"
    >
      <h1
        className="text-dark font-weight-bold"
      >
        Mathemagician
      </h1>
    </ul>
    <ul
      className="nav m-4"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link active text-dark font-weight-bold"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link text-dark font-weight-bolder"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link text-dark font-weight-bold"
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="bg-secondary mt-5 display"
  >
    <h6
      className="m-1 p-4"
    >
      0
    </h6>
  </div>,
  <div
    className="buttonPanel"
  >
    <div
      className="row mt-2 btnPanel"
    >
      <div
        className="col-12"
      >
        <button
          className="button buttons"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button buttons"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button buttons"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button buttons"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="col-12"
      >
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button buttons"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="col-12"
      >
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button buttons"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="col-12"
      >
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button buttons"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="col-12"
      >
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="number buttons"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button button-eql"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
]
`;
